src/API/OnboardingTasks.php

```diff
@@ -715,170 +767,252 @@ function( $has_installed, $bundle ) {
-							'woocommerce-admin'
-						),
-						'actionUrl'   => count( \WC_Shipping_Zones::get_zones() ) > 0
-							? admin_url( 'admin.php?page=wc-settings&tab=shipping' )
-							: null,
-						'actionLabel' => __( "Let's go", 'woocommerce-admin' ),
-						'isComplete'  => count( \WC_Shipping_Zones::get_zones() ) > 0,
-						'isVisible'   => in_array( 'physical', $product_types, true ) ||
-							count(
-								wc_get_products(
-									array(
-										'virtual' => false,
-										'limit'   => 1,
-									)
-								)
-							) > 0,
-						'time'        => __( '1 minute', 'woocommerce-admin' ),
-					),
-					array(
-						'id'         => 'marketing',
-						'title'      => __( 'Set up marketing tools', 'woocommerce-admin' ),
-						'content'    => __(
-							'Add recommended marketing tools to reach new customers and grow your business',
-							'woocommerce-admin'
-						),
-						'isComplete' => $has_installed_marketing_extensions,
-						'isVisible'  => Features::is_enabled( 'remote-free-extensions' ) && count( $marketing_extension_bundles ) > 0,
-						'time'       => __( '1 minute', 'woocommerce-admin' ),
-					),
-					array(
-						'id'          => 'appearance',
-						'title'       => __( 'Personalize my store', 'woocommerce-admin' ),
-						'content'     => __(
-							'Add your logo, create a homepage, and start designing your store.',
-							'woocommerce-admin'
-						),
-						'actionLabel' => __( "Let's go", 'woocommerce-admin' ),
-						'isComplete'  => get_option( 'woocommerce_task_list_appearance_complete' ),
-						'isVisible'   => true,
-						'time'        => __( '2 minutes', 'woocommerce-admin' ),
-					),
+						)
+					) > 0,
+				'time'        => __( '1 minute', 'woocommerce-admin' ),
+			),
+			array(
+				'id'         => 'marketing',
+				'title'      => __( 'Set up marketing tools', 'woocommerce-admin' ),
+				'content'    => __(
+					'Add recommended marketing tools to reach new customers and grow your business',
+					'woocommerce-admin'
 				),
+				'isComplete' => $has_installed_marketing_extensions,
+				'isVisible'  => Features::is_enabled( 'remote-free-extensions' ) && count( $marketing_extension_bundles ) > 0,
+				'time'       => __( '1 minute', 'woocommerce-admin' ),
+			),
+			array(
+				'id'          => 'appearance',
+				'title'       => __( 'Personalize my store', 'woocommerce-admin' ),
+				'content'     => __(
+					'Add your logo, create a homepage, and start designing your store.',
+					'woocommerce-admin'
+				),
+				'actionLabel' => __( "Let's go", 'woocommerce-admin' ),
+				'isComplete'  => get_option( 'woocommerce_task_list_appearance_complete' ),
+				'isVisible'   => true,
+				'time'        => __( '2 minutes', 'woocommerce-admin' ),
 			),
 		);
 
-		return rest_ensure_response( apply_filters( 'woocommerce_admin_onboarding_tasks', $tasks ) );
+		$snoozed_tasks     = get_option( 'woocommerce_task_list_remind_me_later_tasks' );
+		$tasks_with_snooze = array_map(
```

Comment:
I think this could be added via filter to keep the logic down in this method.  See https://github.com/woocommerce/woocommerce-admin/pull/7538

Reply:
Good call, done in 22297e280f1b810ab4484050a54119574913e499 👍🏻